### DISCUSSION TITLE
Why do we need the kotlin coroutines library for JVM/Android but not iOS in v3?

### DISCUSSION THREAD

**Discussion by garretyoder** [coil-kt/coil#2098] [Q&A]
Not linking kotlinx.coroutines-android or swing for JVM will cause an error due to the missing main dispatcher (even though I'm pretty sure it gets looped in by compose multiplatform, since I can call coroutine methods without linking it).

However, on iOS this is not required. Is there a reason why, or an iOS coroutines target I should be linking against?

**Comment by colinrtwhite** [comment]
This seems like a local build issue as `coroutines-android` is [definitely imported](https://github.com/coil-kt/coil/blob/629e1ff5631616d5ee04a0e7d035b6e5d00a1df2/coil-core/build.gradle.kts#L45).

Coil doesn’t import the main dispatcher on desktop is there are two options: javafx and swing. Importing one would prevent all users from using the other
